pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstorecache_test.go

```diff
@@ -0,0 +1,444 @@
+//go:build !windows
+// +build !windows
+
+package sigstore
+
+import (
+	"container/list"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	selectors1 = Item{
+		Key: "signature1",
+		Value: []SelectorsFromSignatures{
+			{
+				Subject:        "spirex1@example.com",
+				Content:        "content1",
+				LogID:          "1111111111111111",
+				IntegratedTime: "1111111111111111",
+			},
+		},
+	}
+
+	selectors2 = Item{
+		Key: "signature2",
+		Value: []SelectorsFromSignatures{
+			{
+				Subject:        "spirex2@example.com",
+				Content:        "content2",
+				LogID:          "2222222222222222",
+				IntegratedTime: "2222222222222222",
+			},
+		},
+	}
+
+	selectors3 = Item{
+		Key: "signature3",
+		Value: []SelectorsFromSignatures{
+			{
+				Subject:        "spirex3@example.com",
+				Content:        "content3",
+				LogID:          "3333333333333333",
+				IntegratedTime: "3333333333333333",
+			},
+		},
+	}
+
+	selectors3Updated = Item{
+		Key: "signature3",
+		Value: []SelectorsFromSignatures{
+			{
+				Subject:        "spirex3@example.com",
+				Content:        "content4",
+				LogID:          "4444444444444444",
+				IntegratedTime: "4444444444444444",
+			},
+		},
+	}
+
+	selectors2Updated = Item{
+		Key: "signature2",
+		Value: []SelectorsFromSignatures{
+			{
+				Subject:        "spirex2@example.com",
+				Content:        "content5",
+				LogID:          "5555555555555555",
+				IntegratedTime: "5555555555555555",
+			},
+		},
+	}
+)
+
+func TestNewCache(t *testing.T) {
+	want := &cacheImpl{
+		size:     3,
+		items:    list.New(),
+		mutex:    sync.RWMutex{},
+		itemsMap: make(map[string]MapItem),
+	}
+	got := NewCache(3)
+	require.Equal(t, want, got, "NewCache() = %v, want %v", got, want)
```

Comment:
same comment that in the another test class, why you required this message? is not regular message enough? regular message will display a rich diff error, and it will already have test name.
the very same comment for all cases